From 09efc25117a103b97e6ee491027cae795048b970 Mon Sep 17 00:00:00 2001
From: Sebastian Bauer <mail@sebastianbauer.info>
Date: Wed, 4 Apr 2018 10:34:04 +0200
Subject: [PATCH 13/15] Use -idirafter instead of -isystem.

This makes sure that the custom include dirs are added at the end of the
search path list. This is required for fixincludes and gcc 6 libstdc++.
---
 gcc/config/rs6000/amigaos.h | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/gcc/config/rs6000/amigaos.h b/gcc/config/rs6000/amigaos.h
index 94fa93c0be047c08987d7acbc7c71413e164014c..fc6c48ab18563b6d89d7d4b68303d1e6f311421d 100644
--- a/gcc/config/rs6000/amigaos.h
+++ b/gcc/config/rs6000/amigaos.h
@@ -175,13 +175,13 @@
 #define MULTILIB_DEFAULTS {"mcrt=clib2"}
 #endif
 
 /* clib2 */
 
 #define CPP_CLIB2_SPEC "\
--isystem %(base_sdk)clib2/include -isystem %(base_sdk)local/clib2/include"
+-idirafter %(base_sdk)clib2/include -idirafter %(base_sdk)local/clib2/include"
 
 #define LIB_SUBDIR_CLIB2_SPEC "%{mcrt=clib2-ts:lib.threadsafe; :lib}%(lib_subdir_type)"
 
 #define LINK_CLIB2_SPEC "\
 -L%(base_sdk)clib2/%(lib_subdir_clib2) \
 -L%(base_gcc)lib/gcc/ppc-amigaos/%(version)/%{mcrt=clib2-ts:clib2-ts; :clib2}/lib%(lib_subdir_type) \
@@ -197,13 +197,13 @@
 %(base_sdk)clib2/%{mcrt=clib2-ts:lib.threadsafe; :lib}" \
                  "%{!msoft-float:%(lib_subdir_type)}/crtend.o"
 
 /* ixemul */
 
 #define CPP_IXEMUL_SPEC "\
--isystem %(base_sdk)ixemul/include -isystem %(base_sdk)local/ixemul/include"
+-idirafter %(base_sdk)ixemul/include -idirafter %(base_sdk)local/ixemul/include"
 
 #define LIB_SUBDIR_IXEMUL_SPEC "lib%(lib_subdir_type)"
 
 #define LINK_IXEMUL_SPEC "\
 -L%(base_sdk)ixemul/%(lib_subdir_ixemul) \
 -L%(base_gcc)lib/gcc/ppc-amigaos/%(version)/ixemul/%(lib_subdir_ixemul) \
@@ -213,13 +213,13 @@
 #define STARTFILE_IXEMUL_SPEC "%(base_sdk)ixemul/%(lib_subdir_ixemul)/crtbegin.o"
 #define ENDFILE_IXEMUL_SPEC "%(base_sdk)ixemul/%(lib_subdir_ixemul)/crtend.o"
 
 /* libnix */
 
 #define CPP_LIBNIX_SPEC "\
--isystem %(base_sdk)libnix/include -isystem %(base_sdk)local/libnix/include"
+-idirafter %(base_sdk)libnix/include -idirafter %(base_sdk)local/libnix/include"
 
 #define LIB_SUBDIR_LIBNIX_SPEC "lib%(lib_subdir_type)"
 
 #define LINK_LIBNIX_SPEC "\
 -L%(base_sdk)libnix/%(lib_subdir_libnix) \
 -L%(base_gcc)lib/gcc/ppc-amigaos/%(version)/libnix/%(lib_subdir_libnix) \
@@ -228,13 +228,13 @@
 #define STARTFILE_LIBNIX_SPEC "%(base_sdk)libnix/%(lib_subdir_libnix)/crtbegin.o"
 #define ENDFILE_LIBNIX_SPEC "%(base_sdk)libnix/%(lib_subdir_libnix)/crtend.o"
 
 /* newlib */
 
 #define CPP_NEWLIB_SPEC "\
--isystem %(base_sdk)newlib/include -isystem %(base_sdk)local/newlib/include"
+-idirafter %(base_sdk)newlib/include -idirafter %(base_sdk)local/newlib/include"
 
 #define LIB_SUBDIR_NEWLIB_SPEC "lib%(lib_subdir_type)"
 
 #define LINK_NEWLIB_SPEC "\
 -L%(base_sdk)newlib/%(lib_subdir_newlib) \
 -L%(base_gcc)lib/gcc/ppc-amigaos/%(version)/newlib/%(lib_subdir_newlib) \
@@ -254,15 +254,15 @@
 %{mcrt=clib2|mcrt=clib2-ts: %(cpp_clib2); \
 mcrt=ixemul: %(cpp_ixemul); \
 mcrt=libnix: %(cpp_libnix); \
 mcrt=newlib: %(cpp_newlib); \
 mcrt=default|!mcrt=*: %{mcrt=default|!nostdinc: %(cpp_amiga_default)}; \
 : %eInvalid C runtime library} \
--isystem %(base_sdk)include/include_h \
--isystem %(base_sdk)include/netinclude \
--isystem %(base_sdk)local/common/include \
+-idirafter %(base_sdk)include/include_h \
+-idirafter %(base_sdk)include/netinclude \
+-idirafter %(base_sdk)local/common/include \
 %{mbaserel: %{msdata|msdata=default|msdata=sysv: %e-mbaserel and -msdata options are incompatible}} \
 %{newlib: %e-newlib is obsolete, use -mcrt=newlib instead}"
 
 #undef LINK_SPEC
 #define LINK_SPEC "\
 --defsym __amigaos4__=1 \
-- 
2.11.0

